framework/components/blockchain/blockchain.go

```diff
@@ -21,6 +21,10 @@ type Input struct {
 	// publickey to mint when solana-test-validator starts
 	PublicKey    string `toml:"public_key"`
 	ContractsDir string `toml:"contracts_dir"`
+	// programs to deploy on solana-test-validator start
+	// a map of program name to program id
+	// there needs to be a matching .so file in contracts_dir
```

Comment:
matching meaning `<program_name.so>`?

Reply:
Yeah, 
```
[solana_config.solana_programs]
mcm = "6UmMZr5MEqiKWD5jqTJd1WCR5kT8oZuFYBLJFi1o6GQX"
```

expects the file `<contracts_dir>/mcm.so` and it will use `6UmMZr5MEqiKWD5jqTJd1WCR5kT8oZuFYBLJFi1o6GQX` as the program id , so it can be looked up later for testing.

> also, should programs reside in the root of contracts_dir or in <contracts_dir>/programs?

No strong opinions, i think contracts are programs in solana, but i was basically following the convention in [chainlink-ccip](https://github.com/smartcontractkit/chainlink-ccip/blob/609f7b0c9734b3cfc1d1a1aea0fd1ddf4c90bd0c/chains/solana/contracts/tests/testutils/anchor.go#L72-L72)